bases/renku_data_services/background_jobs/core.py

```diff
@@ -264,3 +264,38 @@ async def migrate_user_namespaces_make_all_public(config: SyncConfig) -> None:
         )
         await authz.client.WriteRelationships(authz_change)
         logger.info(f"Made user namespace {ns_id} public")
+
+
+async def migrate_storages_v2_to_data_connectors(config: SyncConfig) -> None:
+    """Move storages_v2 to data_connectors."""
+    logger = logging.getLogger("background_jobs").getChild(migrate_storages_v2_to_data_connectors.__name__)
+
+    api_user = InternalServiceAdmin(id=ServiceAdminId.migrations)
+    storages_v2 = await config.data_connector_migration_tool.get_storages_v2(requested_by=api_user)
+
+    if not storages_v2:
+        logger.info("Nothing to do.")
+        return
+
+    logger.info(f"Migrating {len(storages_v2)} cloud storage v2 items to data connectors.")
+    failed_storages: list[str] = []
+    first_error: Exception | None = None
+    for storage in storages_v2:
+        try:
+            data_connector = await config.data_connector_migration_tool.migrate_storage_v2(
+                requested_by=api_user, storage=storage
+            )
+            logger.info(f"Migrated {storage.name} to {data_connector.namespace.slug}/{data_connector.slug}.")
+            logger.info(f"Deleted storage_v2: {storage.storage_id}")
+        except Exception as err:
+            logger.error(f"Failed to migrate {storage.name}.")
+            logger.error(err)
+            failed_storages.append(str(storage.storage_id))
+            if first_error is None:
+                first_error = err
+
+    logger.info(f"Migrated {len(storages_v2)-len(failed_storages)}/{len(storages_v2)} data connectors.")
+    if failed_storages:
+        logger.error(f"Migration failed for storages: {failed_storages}.")
+    if first_error is not None:
+        raise first_error
```

Comment:
issue: I don't think we want to raise here. We had it in the past where some migration raised an error, which caused the job to fail without doing latter migrations. `fix_mismatched_project_namespace_ids` failed and so the migrating groups jobs were never executed which had caused issues until we rolled out a fix a week later. The ugly quick fix back then was to just not raise when that error happened, but the whole background jobs code is still kinda brittle in this regard.

This also gets a bit tricky to manage in cases where a latter migration should only be done if a previous one passed, though I don't think we have such dependencies at the moment?

In any case, I think we should collect exceptions for each of the tasks in the short_period_sync job and only raise at the end once all things have been done. This does not mean you need to do that in this PR for all the migration tasks, if you don't want to, a follow-up issue is fine, but until we have a proper solution we should not raise in a migration task directly.

And I know that right now it doesn't matter since this is the last migration in the list, but for sure the next person along will just add another one after this one without considering it.

Reply:
Done in dcc3080584e3618b7b41e67a19a51402e6771b4b.